src/OpenSpaceToolkit/Astrodynamics/Trajectory/State/CoordinatesSubset.cpp

```diff
@@ -54,6 +55,49 @@ Size CoordinatesSubset::getSize() const
     return size_;
 }
 
+VectorXd CoordinatesSubset::add(
+    [[maybe_unused]] const Instant& anInstant,
+    const VectorXd& aFullCoordinatesVector,
+    const VectorXd& anotherFullCoordinatesVector,
+    [[maybe_unused]] const Shared<const Frame>& aFrameSPtr,
+    const Shared<const CoordinatesBroker>& aCoordinatesBrokerSPtr
+) const
+{
+    return aCoordinatesBrokerSPtr->extractCoordinates(aFullCoordinatesVector, *this) +
+           aCoordinatesBrokerSPtr->extractCoordinates(anotherFullCoordinatesVector, *this);
+}
+
+VectorXd CoordinatesSubset::subtract(
+    [[maybe_unused]] const Instant& anInstant,
+    const VectorXd& aFullCoordinatesVector,
+    const VectorXd& anotherFullCoordinatesVector,
+    [[maybe_unused]] const Shared<const Frame>& aFrameSPtr,
+    const Shared<const CoordinatesBroker>& aCoordinatesBrokerSPtr
+) const
+{
+    return aCoordinatesBrokerSPtr->extractCoordinates(aFullCoordinatesVector, *this) -
+           aCoordinatesBrokerSPtr->extractCoordinates(anotherFullCoordinatesVector, *this);
+}
+
+VectorXd CoordinatesSubset::inFrame(
+    [[maybe_unused]] const Instant& anInstant,
+    const VectorXd& aFullCoordinatesVector,
+    [[maybe_unused]] const Shared<const Frame>& fromFrame,
+    [[maybe_unused]] const Shared<const Frame>& toFrame,
+    const Shared<const CoordinatesBroker>& aCoordinatesBrokerSPtr
+) const
+{
+    VectorXd coordinates = aCoordinatesBrokerSPtr->extractCoordinates(aFullCoordinatesVector, *this);
+
+    return coordinates;
+}
+
+Shared<const CoordinatesSubset> CoordinatesSubset::Mass()
+{
+    static const Shared<const CoordinatesSubset> mass = std::make_shared<CoordinatesSubset>("MASS", 1);
```

Comment:
Does the mass value need to be static? I wonder if this could lead to weird behavior if you ever try to define two different `MASS` coordinate subsets.

Also might be worth removing the explicit `add` functions from other classes. Lemme know what you think about this:
https://github.com/open-space-collective/open-space-toolkit-astrodynamics/pull/202

Reply:
> Also might be worth removing the explicit add functions from other classes. Lemme know what you think about this

Absolutely 🙏

Reply:
So the reason this is static, is because we want to only create 1 pointer, and everytime this function is called, you just return the pointer to this (and thus increment the ownership by 1). Rather than creating a _new_ pointer and then allocating memory on the heap again. This is similar to how the Frame::GCRF() etc. works.